.github/workflows/docker.yml

```diff
@@ -0,0 +1,54 @@
+name: Build & Push
+#  Build & Push rebuilds the iavl docker image on every push to master and creation of tags
+# and pushes the image to https://hub.docker.com/r/interchainio/iavl/tags
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+" # Push events to matching v*, i.e. v1.0, v20.15.10
+      - "v[0-9]+.[0-9]+.[0-9]+-rc*" # Push events to matching v*, i.e. v1.0-rc1, v20.15.10-rc5
```

Comment:
I think these will only match versions with major.minor.patch, not for example, major.minor. Correct me if I'm wrong. Why not just `v[0-9]+.[0-9]*.?[0-9]*.*` or even more relaxed `v[0-9]+.*`?

Reply:
```
v[12].[0-9]+.[0-9]+ | Matches all semantic versioning tags with major version 1 or 2
```

Went with this approach from github actions. We could relax it more